Use direct access for rbd backed images (bsc#945043)

dirkmueller opened the pull request.

dirkmueller committed:
Switch to Glance V2 API

This allows performance optimisations by being able to access
the backend location in the glance image store, and should be
not causing any downside.

dirkmueller committed:
Enable direct backend access for rbd (bsc#945043)

This allows a more performant COW initialisation of a volume from
a snapshot stored in glance.

This is related to https://github.com/crowbar/crowbar-openstack/pull/110
on master, however its not a straight forward backport but only enables
the rbd backend.

vuntz commented on chef/cookbooks/cinder/templates/default/cinder.conf.erb:
```diff
@@ -740,7 +740,10 @@ nova_api_insecure=<%= @nova_api_insecure %>
 # the direct_url.  Currently supported schemes: [file]. (list
 # value)
 #allowed_direct_url_schemes=
-
```
Can we keep that empty line?

vuntz commented on chef/cookbooks/cinder/templates/default/cinder.conf.erb:
```diff
@@ -740,7 +740,10 @@ nova_api_insecure=<%= @nova_api_insecure %>
+<% end -%>
 
 #
 # Options defined in cinder.image.image_utils
```
In the other PR, we have https://github.com/crowbar/crowbar-openstack/pull/110/files#diff-4e35a687f5956e18c68d43a410d78e78R1023 (`image_upload_use_cinder_backend = True` and `image_upload_use_internal_tenant = False`). This is not available in juno?

dirkmueller replied:
Thats correct, the cinder integration in Glance is a new feature in Kilo

vuntz commented:
Could you mention it's a backport of https://github.com/crowbar/crowbar-openstack/pull/110 in the commit message?

dirkmueller committed:
Avoid hang in accept on handshake (bsc#949241)

Workaround this by enabling the security fix for the slowlaris
attack. see https://bugs.launchpad.net/nova/+bug/1361360 for
the full story.

vuntz commented:
+1

dirkmueller commented:
@vuntz done

vuntz commented:
Don't know what you did, but still +1 :-)

dirkmueller commented:
@vuntz updated the git commit message and fixed the missing whitespace..

aplanas commented:
+1

toabctl commented:
+1
